kafka-client/src/main/java/dev/responsive/kafka/clients/ResponsiveConsumer.java

```diff
@@ -22,69 +22,56 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
-import java.util.Optional;
 import java.util.regex.Pattern;
 import org.apache.kafka.clients.consumer.Consumer;
 import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
 import org.apache.kafka.clients.consumer.OffsetAndMetadata;
 import org.apache.kafka.clients.consumer.OffsetCommitCallback;
 import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.utils.LogContext;
 import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class ResponsiveConsumer<K, V> extends DelegatingConsumer<K, V> {
   private final List<Listener> listeners;
-  private final Logger logger;
+  private final Logger log;
 
   private static class RebalanceListener implements ConsumerRebalanceListener {
-    private final Optional<ConsumerRebalanceListener> wrappedRebalanceListener;
+    private final ConsumerRebalanceListener wrappedRebalanceListener;
     private final List<Listener> listeners;
-    private final Logger logger;
-
-    public RebalanceListener(final List<Listener> listeners, final Logger logger) {
-      this(Optional.empty(), listeners, logger);
-    }
+    private final Logger log;
 
     public RebalanceListener(
         final ConsumerRebalanceListener wrappedRebalanceListener,
         final List<Listener> listeners,
-        final Logger logger
-    ) {
-      this(Optional.of(wrappedRebalanceListener), listeners, logger);
-    }
-
-    private RebalanceListener(
-        final Optional<ConsumerRebalanceListener> wrappedRebalanceListener,
-        final List<Listener> listeners,
-        final Logger logger
+        final Logger log
     ) {
       this.wrappedRebalanceListener = wrappedRebalanceListener;
       this.listeners = listeners;
-      this.logger = logger;
+      this.log = log;
     }
 
     @Override
     public void onPartitionsLost(final Collection<TopicPartition> partitions) {
-      wrappedRebalanceListener.ifPresent(l -> l.onPartitionsLost(partitions));
       for (final var l : listeners) {
-        ignoreException(logger, () -> l.onPartitionsLost(partitions));
+        ignoreException(log, () -> l.onPartitionsLost(partitions));
       }
+      wrappedRebalanceListener.onPartitionsLost(partitions);
```

Comment:
Changed the order here to make sure we always invoke our listeners, since the rebalance callbacks may occasionally throw exceptions. I figured this makes sense since if, for example, we keep hitting `onPartitionsLost` over and over because it keeps throwing some exception, then we'll want to know about it from our listeners here

Reply:
Can you leave a comment to this effect and add a test?